src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionMiddleware.cs

```diff
@@ -94,28 +97,41 @@ public Task Invoke(HttpContext context)
             context.Response.StatusCode = _statusCode;
             context.Response.Headers[HeaderNames.Location] = redirectUrl;
 
+            _logger.LogDebug($"Redirecting to https.");
+
             return Task.CompletedTask;
         }
 
-        private void CheckForHttpsPorts()
+        private bool TryGetHttpsPort(out int port)
         {
             // The IServerAddressesFeature will not be ready until the middleware is Invoked,
             // Order for finding the HTTPS port:
             // 1. Set in the HttpsRedirectionOptions
             // 2. HTTPS_PORT environment variable
             // 3. IServerAddressesFeature
-            // 4. 443 (or not set)
+            // 4. Fail if not set
+
+            if (_portEvaluated)
+            {
+                port = _httpsPort ?? -1;
```

Comment:
Set port initially to -1 to avoid duplication.

Reply:
?